frontend/src/components/Explore.jsx

```diff
@@ -0,0 +1,204 @@
+import { useState, useEffect } from "react";
+import "./Explore.css";
+import SearchBar from "./SearchBar";
+import Workout from "./Workout";
+import Meal from "./Meal";
+import Trainer from "./Trainer";
+import Modal from "./Modal";
+import { onAuthStateChanged } from "firebase/auth";
+import { auth, generateContent } from "../firebase";
+import DisplayWorkout from "./DisplayWorkout";
+import DisplayMeal from "./DisplayMeal";
+
+function Explore() {
+  const [user, setUser] = useState();
+  const [workouts, setWorkouts] = useState([]);
+  const [meals, setMeals] = useState([]);
+  const [modal, setModal] = useState(false);
+  const [refresh, setRefresh] = useState(0);
+  const [workoutMeal, setWorkoutMeal] = useState(true); //true == workout, false == meal
+  const [chatting, setChatting] = useState(false);
+  const [currentWorkout, setCurrentWorkout] = useState([]);
+  const [currentMeal, setCurrentMeal] = useState([]);
+
+  useEffect(() => {
+    onAuthStateChanged(auth, (prof) => {
+      if (prof) {
+        // User is signed in, see docs for a list of available properties
+        // https://firebase.google.com/docs/reference/js/firebase.User
+
+        fetch(`http://localhost:3000/workouts`)
+          .then((response) => {
+            if (!response.ok) {
+              throw new Error(`HTTP error! status: ${response.status}`);
+            }
+            return response.json(); // Parse JSON data from the response
+          })
+          .then((data) => {
+            // Handle successful response
+            setWorkouts(data);
+          })
+          .catch((error) => {
+            console.error("Error fetching boards:", error);
+          });
```

Comment:
These functions are basically the same. Is it possible to create an abstract function, and pass in URL and other functions like setWorkouts(), and reuse the abstract function?

Reply:
I can look into making a http-master function for efficiency!